File: December - 08/Solution.java
import java.util.*;
class Solution {
    public int countStudents(int[] students, int[] sandwiches) {

        Stack<Integer> stack = new Stack<>();
        for (int i = sandwiches.length - 1; i >= 0; i--) {
            stack.push(sandwiches[i]);
        }

        ArrayDeque<Integer> queue = new ArrayDeque<>();
        for (int i : students) queue.offerLast(i);

        int rotations = 0;

        while (!stack.isEmpty() && !queue.isEmpty()) {
            int stackTop = stack.peek();
            int queueTop = queue.peekFirst();

            if (stackTop == queueTop) {
                stack.pop();
                queue.pollFirst();
                rotations = 0; 
            } else {
                queue.offerLast(queue.pollFirst());
                rotations++;
            }

            if (rotations == queue.size()) {
                return stack.size();
            }
        }

        return 0;
    }
}

File: December - 08/Solution.py
def countStudents(students, sandwiches):
    count = {0: 0, 1: 0}
    for s in students:
        count[s] += 1
    for sand in sandwiches:
        if count[sand] == 0:
            break
        count[sand] -= 1
    return count[0] + count[1]
students = list(map(int, input().split()))
sandwiches = list(map(int, input().split()))
print(countStudents(students, sandwiches))
